packages/client/src/providers/AuthProvider/index.js

```diff
@@ -32,31 +25,68 @@ const useAuth = () => {
     return useContext(authContext);
 }
 
+// TODO: possibly create a enum struct or flag
+export const LOGGED_OUT = "LOGGED_OUT";
+
 const useAuthProvider = () => {
     const [user, setUser] = useState(null);
     const [initialized, setInitialized] = useState(false);
 
-    const signup = (email, password) => {
-        return firebase.auth().createUserWithEmailAndPassword(email, password);
-    }
+    /**
+     * Signs a user up.
+     * @param {string} email
+     * @param {string} password
+     * @param {Mentor|Parent} user object
+     *
+     * @return {Promise<void>} indicates successful account creation.
+     */
+    const signup = (email, password, user) => {
+        return Auth.createUserWithEmailAndPassword(email, password)
+            .then(() => {
+                return user.create(Auth.currentUser.uid);
```

Comment:
is `Auth.currentUser` guaranteed to be set correctly when this is called?

Reply:
Ya it is.